src/components/PublishCarForm.component.tsx

```diff
@@ -0,0 +1,306 @@
+import React, { useEffect } from "react";
+import { SubmitHandler, useForm } from "react-hook-form";
+import { toast } from "react-toastify";
+import "react-toastify/dist/ReactToastify.css";
+import styled from "styled-components";
+
+import {
+  useCreateCarMutation,
+  useFormDataQuery,
+  useModelsLazyQuery,
+} from "../graphql/generated/graphql";
+import { modelsbyBrandIdVariables } from "../graphql/variables";
+import { GlobalStyle, defaultTheme } from "../theme";
+import { ConditionFieldset } from "./ConditionFieldset";
+import { FormSelectInput } from "./FormSelectInput.component";
+import {
+  ButtonOnHoverOppacity,
+  CSSFlexCol,
+  FlexRow,
+  Form,
+  H2,
+  H3,
+  InputStyle,
+  P,
+  ValidationText,
+} from "./styled";
+import Loader from "./styled/Loader.component";
+
+interface FormValues {
+  brand_id: number;
+  model_id?: number;
+  year?: number;
+  state_id?: number;
+  city_id?: number;
+  vin?: string;
+  color_id?: number;
+  odometer: number;
+  condition: string;
+  sale_date: string;
+  price: number;
+  title?: string;
+}
+const VALIDATION_MESSAGES = {
+  BRAND: "Select your car's Brand",
+  MODEL: "Select your car's Model",
+  YEAR: "Enter your car's fabrication year",
+  CITY: "Select your car's city of procedence",
+  VIN: "Enter your Vehicle's Identification Number",
+  COLOR: "Select your car's color",
+  CONDITION: "Pick your car's condition",
+  SALE_DATE: "Select when you want to sell your car",
+  PRICE: "Enter a price for your car",
+};
+export const PublishCarForm = () => {
+  const { data: formsData, error: dataError, loading } = useFormDataQuery();
+  const [
+    fetchModels,
+    { data: modelsData, error: modelsError, loading: modelsLoading },
+  ] = useModelsLazyQuery();
+
+  const [createCar, { data: mutationReturn, loading: mutationLoading }] =
+    useCreateCarMutation();
+  const {
+    register,
+    reset,
+    handleSubmit,
+    watch,
+    setValue,
+    getValues,
+    formState: { errors },
+  } = useForm<FormValues>();
+
+  const threeMonthsAhead = new Date();
+  threeMonthsAhead.setDate(threeMonthsAhead.getDate() + 90);
+
+  const brandId = watch("brand_id");
```

Comment:
I suggest to create a new component to handle request and changes from the `brand` select instead of handling this on the Publish form, to me this component should be in charge of just collecting, validating and sending the data